docs/source/app-dev/app-arch.rst

```diff
@@ -1,295 +1,120 @@
++--------------------------------------------------------------+--------------------------------------------------------------------------+
+| Name                                                         | Summary                                                                  |
++==============================================================+==========================================================================+
+| `@daml/react <https://www.npmjs.com/package/@daml/react>`_   | React hooks to query/create/exercise DAML contracts                      |
++--------------------------------------------------------------+--------------------------------------------------------------------------+
+| `@daml/ledger <https://www.npmjs.com/package/@daml/ledger>`_ | DAML ledger object to connect and directly submit commands to the ledger |
++--------------------------------------------------------------+--------------------------------------------------------------------------+
+| `@daml/types  <https://www.npmjs.com/package/@daml/types>`_  | DAML data types                                                          |
++--------------------------------------------------------------+--------------------------------------------------------------------------+
 
-Examples of these are automation and interactive applications that let a user or code react to complex state on the ledger (e.g. the DA Navigator tool).
+You can install any of these libraries by running ``yarn add @daml/react`` in the ``ui`` directory
+of your project. Please explore the ``create-daml-app`` example project to see the usage of these
+libraries.
 
-Which approach to take
-======================
+The interface between the frontend and the backend of your application is defined by your DAML code.
+To make your life easy when interacting with the ledger, the DAML assistant can generate
+corresponding typescript data definitions for the data types declared in DAML.
 
-For all except the simplest applications, we generally recommend the state-driven approach. State-driven applications are easier to reason about when determining correctness, so this makes design and implementation easier.
+.. code-block:: bash
 
-In practice, most applications are actually a mixture of the two styles, with one predominating. It is easier to add some event handling to a state-driven application, so it is better to start with that style.
+  daml codegen ts .daml/dist/<your-project-name.dar> -o daml2ts
 
-Structuring an application
-**************************
+This will create a typescript project under ``daml2ts`` that you can install by changing directory
+to ``daml2ts`` and running ``yarn install``. We recommend using the provided ``daml-start.sh``
+script at the root of the project directory of ``create-daml-app``. This will take care of creating
+and updating the ``daml2ts`` library whenever you change your DAML model automatically.
```

Comment:
Are the two (yarn install and daml-start.sh) options or I need to do both, i.e., do I need to do yarn install and daml-start.sh? If I can use only create-daml-app/daml-start.sh I would only describe that case. Ideally this would be all done in the background (typescript codegen) when the user uses daml-start.sh.

Reply:
you need to do both, the process is not that smooth at this point.